src/azure-cli-core/azure/cli/core/auth/msal_authentication.py

```diff
@@ -38,17 +38,20 @@ def __init__(self, client_id, username, **kwargs):
         """
         super().__init__(client_id, **kwargs)
 
-        accounts = self.get_accounts(username)
+        # Make sure username is specified, otherwise MSAL returns all accounts
+        assert username, "username must be specified, got {!r}".format(username)
 
-        if not accounts:
-            raise CLIError("User {} does not exist in MSAL token cache. Run `az login`.".format(username))
+        accounts = self.get_accounts(username)
 
+        # MSAL should never return multiple accounts for the same username
```

Comment:
This statement is not strictly true. [In MSAL, an account's `username` is primarily derived by the `preferred_username` claim from the ID token](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/msal/token_cache.py#L193), which is [not guaranteed to be unique](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#payload-claims). I do like the Az CLI error message, though.

```suggestion
        # Usernames are usually unique. We are collecting corner cases to better understand its behavior.
```